api-documentation.md

```diff
@@ -2,41 +2,66 @@
 
 > Note: using [this markdown template](https://gist.github.com/iros/3426278).
 
-## **List links**
+## **GET links**
 
 Retrieve a list of links ordered by creation date.
 
-* **URL**:  `?do=api[TODO]`
+* **URL**: Endpoint discussion in #3.
 * **Method:** `GET`
-*  **URL Params**
+* **Path Parameters:** none
+* **URL Parameters**
 
    **Optional:**
  
     * `offset=[numeric]`
       Offset from which to start listing links (default: 0).
     * `limit=[numeric]`
-      Number of links to retrieve (default ?) or `all`.
-      
+      Number of links to retrieve (default 20) or `all`.
+    * `searchterm=[string]`
+      Search terms across all links fields.
+    * `searchtags=[string]`
+      Search for specifics tags. Tag list should be separated by a `+` delimitor.
+* **Body Parameters:** none
 
-* **Body Params:** none
-
-* **Success Response:**
-* **Content:** 
+* **Success Response:** `200`
+  **Content:** 
     ```json
     [
       { 
-        "todo": "link obj"
+        "id": "linkID",
+        "url": "Shaared URL",
+        "title": "link title",
+        "description": "link description",
+        "tags": [
+          "shaarli",
+          "php",
+          "api"
+        ],
+        "private": true,
+        "created": "2016-06-15T19:23:14+0200",
```

Comment:
> does description need a (optional) mimetype nowadays (plaintext vs. markdown)?

Markdown content is also plaintext. Instead of adding an additional field, a service returning settings, included enabled plugins is probably better. It would let clients interpret any plugin if they want to.

> what is "todo"?

I had **to do** this object definition. :stuck_out_tongue: